The cgroup name can't be nil while systemReserved is designated

FengyunPan2 opened the pull request:
related to #83228
Currently kubelet just check 'systemReserved' and 'kubeReserved'
parameters, did not check cgourp name. This pr would check cgroup
name before enforcing System reserved on cgroup system.slice

/kind bug

**Release note**:
```release-note
NONE
```

FengyunPan2 commented:
/sig kubelet

tedyu commented on pkg/kubelet/cm/node_container_manager_linux.go:
```diff
@@ -102,34 +102,47 @@ func (cm *containerManagerImpl) enforceNodeAllocatableCgroups() error {
-			cm.recorder.Event(nodeRef, v1.EventTypeWarning, events.FailedNodeAllocatableEnforcement, message)
-			return fmt.Errorf(message)
+		if len(nc.SystemReservedCgroupName) == 0 {
+			klog.Warningf("Warning: when %s is enabled on enforceNodeAllocatable, both systemReserved and systemReservedCgroup should be designated,"+
```
I think an error should be returned here.

FengyunPan2 replied:
Ok, update it now.

tedyu commented on pkg/kubelet/cm/node_container_manager_linux.go:
```diff
@@ -111,6 +118,13 @@ func (cm *containerManagerImpl) enforceNodeAllocatableCgroups() error {
 		cm.recorder.Eventf(nodeRef, v1.EventTypeNormal, events.SuccessfulNodeAllocatableEnforcement, "Updated limits on system reserved cgroup %v", nc.SystemReservedCgroupName)
 	}
 	if nc.EnforceNodeAllocatable.Has(kubetypes.KubeReservedEnforcementKey) {
+		if len(nc.KubeReservedCgroupName) == 0 {
```
Maybe extract the body of if block to a method so that it can be used for line 105 as well.

FengyunPan2 replied:
Done

FengyunPan2 committed:
The cgroup name can't be nil while systemReserved is designated

Related to: #83228
Currently kubelet just check 'systemReserved' and 'kubeReserved'
parameters, did not check cgourp name. This pr would check cgroup
name before enforcing System reserved on cgroup system.slice

FengyunPan2 commented:
@neolit123 Yes, this PR is useful for users, introduce a user-facing change.

neolit123 commented:
> Yes, this PR is useful for users, introduce a user-facing change.

please add the note.
see an example here https://github.com/kubernetes/kubernetes/pull/83694

FengyunPan2 commented:
/test pull-kubernetes-e2e-kind
/test pull-kubernetes-kubemark-e2e-gce-big

dims commented:
/uncc
